space_age.py

```diff
@@ -0,0 +1,37 @@
+class SpaceAge(object):
+    SECONDS_IN_EARTH_YEAR = 60 * 60 * 24 * 365
+	
+    def __init__(self, seconds):
+        self.seconds = seconds
+        self.planets = {
+            'Earth': 1, 'Mercury': 0.2408467, 'Venus': 0.61519726,
+            'Mars': 1.8808158, 'Jupiter': 11.862615, 'Saturn': 29.447498,
+            'Uranus': 84.016846, 'Neptune': 164.79132
+        }
+        
+    def on_mercury(self):
+       return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Mercury']
+
+    def on_venus(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Venus']
+
+    def on_earth(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Earth']
+
+    def on_mars(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Mars']
+
+    def on_jupiter(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Jupiter']
+
+    def on_saturn(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Saturn']
+
+    def on_uranus(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Uranus']
+
+    def on_neptune(self):
+        return self.seconds/SECONDS_IN_EARTH_YEAR*self.planets['Neptune']
+
+
+SpaceAge.on_neptune(71997456)
```

Comment:
@zarniak The test for this invoke your code like that `SpaceAge(2134835688).on_mercury()`. So that's the way we were aiming for your code to work.

Some hints about what's happening here:
- `SpaceAge(71997456)` creates an object which can be re-used later. Technically, `__init__` defines what happens with each object when it is created. This way you could pre-compute the age on all planets in the `__init__` and return the results each time `on_...` gets called.
- That first `self` that is presents in each method's signature is a technical artifact that allows the method to have access to the object that is calling it. TBH I don't know why it was designed so that it has to be explicitly mentioned :man_shrugging: